drake/systems/frames/CoordinateFrame.m

```diff
@@ -549,13 +550,39 @@ function setupLCMOutputs(obj,mdl,subsys,subsys_portnum,options)
     function coordinates = generateDefaultCoordinates(prefix)
       % generates the default coordinate labels for a coordinate Frame
       % when no value is passed in, based on the prefixes provided.
-      ind=1;
-      [dim, ~] = size(prefix);
-      function str=coordinateName(~)
-        str=[prefix(ind),sprintf('%d',sum(prefix(1:ind)==prefix(ind)))];
-        ind=ind+1;
+      
+      % specifically, we take the prefix string corresponding to an array of
+      % characters, and create a cell array with a unique identifier
+      % appended to each string.
+      
+      % for example, if s='aabab', 
+      % coordinates = {'a1', 'a2', 'b1', 'a3', 'b2'}
+      
+      % initialize our empty cell array
+      n = length(s);
```

Comment:
Yes, it looks like I must have made an error when refactoring the names.
Thank you for the catch.
On 8 Jun 2016 09:28, "Chien-Liang Fok" <notifications@github.com> wrote:

> In drake/systems/frames/CoordinateFrame.m
> <https://github.com/RobotLocomotion/drake/pull/2420#discussion_r66289960>:
>
> > @@ -549,13 +550,39 @@ function setupLCMOutputs(obj,mdl,subsys,subsys_portnum,options)
> >      function coordinates = generateDefaultCoordinates(prefix)
> >        % generates the default coordinate labels for a coordinate Frame
> >        % when no value is passed in, based on the prefixes provided.
> > -      ind=1;
> > -      [dim, ~] = size(prefix);
> > -      function str=coordinateName(~)
> > -        str=[prefix(ind),sprintf('%d',sum(prefix(1:ind)==prefix(ind)))];
> > -        ind=ind+1;
> > +
> > +      % specifically, we take the prefix string corresponding to an array of
> > +      % characters, and create a cell array with a unique identifier
> > +      % appended to each string.
> > +
> > +      % for example, if s='aabab',
> > +      % coordinates = {'a1', 'a2', 'b1', 'a3', 'b2'}
> > +
> > +      % initialize our empty cell array
> > +      n = length(s);
>
> Should s be prefix?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/2420/files/6ccc4a0f7dfe529d1db8ea0b359be2da064f3da9#r66289960>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFqHEZ3y1_-pvaIuJYzjdxJbHNZey6wrks5qJu2mgaJpZM4Ijmgc>
> .
>